### PR TITLE
Remove unnecessary GetIndexer

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?

/kind cleanup
/sig scheduling

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -499,7 +499,9 @@ func newPodInformer(cs clientset.Interface, resyncPeriod time.Duration) cache.Sh
 	tweakListOptions := func(options *metav1.ListOptions) {
 		options.FieldSelector = selector
 	}
-	informer := coreinformers.NewFilteredPodInformer(cs, metav1.NamespaceAll, resyncPeriod, cache.Indexers{}, tweakListOptions)
+	informer := coreinformers.NewFilteredPodInformer(cs, metav1.NamespaceAll, resyncPeriod, cache.Indexers{
+		cache.NamespaceIndex: cache.MetaNamespaceIndexFunc,
+	}, tweakListOptions)
```
Comment (chendave) [inline pkg/scheduler/scheduler.go:504]: any reason for this change? `defaultInformer` has already has the `Indexers` set. 

```golang
func (f *podInformer) defaultInformer(client kubernetes.Interface, resyncPeriod time.Duration) cache.SharedIndexInformer {
	return NewFilteredPodInformer(client, f.namespace, resyncPeriod, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, f.tweakListOptions)
}

func (f *podInformer) Informer() cache.SharedIndexInformer {
	return f.factory.InformerFor(&corev1.Pod{}, f.defaultInformer)
}
```

besides, pls note the comment here
> // The PodInformer allows indexers to be added, but note that only non-conflict indexers are allowed.

**Review by chendave — COMMENTED** [review]
Just a question.

pls unhold if this is ready to go.

File: pkg/scheduler/scheduler.go (hunk 1 above)
Comment (kerthcet) [reply pkg/scheduler/scheduler.go]: That's definitely a mistake, thanks for pointing this out. But we're using a shared informer here, not the default informer. 
Reverted, PTAL @chendave

**Review by chendave — COMMENTED** [review]
nice cleanup!

**Pull request merged**